src/v/cluster/tm_stm.cc

```diff
@@ -62,10 +63,102 @@ tm_stm::tm_stm(
+
+    auto ns_label = sm::label("namespace");
+    auto topic_label = sm::label("topic");
+    auto partition_label = sm::label("partition");
+
+    const auto& ntp = _raft->ntp();
+    const std::vector<sm::label_instance> labels = {
+      ns_label(ntp.ns()),
+      topic_label(ntp.tp.topic()),
+      partition_label(ntp.tp.partition()),
+    };
+
+    auto setup = [this, &labels]<typename MetricDef, bool internal>(
+                   const std::vector<sm::label>& aggregate_labels) {
+        std::vector<MetricDef> defs;
+
+        defs.emplace_back(
+          sm::make_gauge(
+            "open_transactions",
+            [this] { return _probe.open_transactions; },
+            sm::description(
+              "Number of open transactions, not committed/aborted."),
+            labels)
+            .aggregate(aggregate_labels));
+
+        defs.emplace_back(
+          sm::make_counter(
+            "committed_transactions",
+            [this] { return _probe.committed_transactions; },
+            sm::description("Number of committed transactions."),
+            labels)
+            .aggregate(aggregate_labels));
+
+        defs.emplace_back(sm::make_counter(
+                            "aborted_transactions",
+                            [this] { return _probe.aborted_transactions; },
+                            sm::description("Number of aborted transactions."),
+                            labels)
+                            .aggregate(aggregate_labels));
+
+        if constexpr (internal) {
+            defs.emplace_back(
+              sm::make_histogram(
+                "participants_per_transaction",
+                [this] {
+                    return _probe.participants_per_transaction
+                      .tx_participants_histogram_logform();
+                },
+                sm::description("Number of partitions and groups participating "
+                                "in a transaction."),
+                labels)
+                .aggregate(aggregate_labels));
+        }
+        defs.emplace_back(
+          sm::make_gauge(
+            "active_transaction_ids",
+            [this] { return _transactions.size(); },
+            sm::description(
+              "Number of transaction ids active on the coordinator."),
+            labels)
+            .aggregate(aggregate_labels));
+        defs.emplace_back(
+          sm::make_counter(
+            "expired_transaction_ids",
+            [this] { return _probe.expired_transaction_ids; },
+            sm::description(
+              "Number of transaction ids that got garbage collected."),
+            labels)
+            .aggregate(aggregate_labels));
+        return defs;
+    };
+    if (!config::shard_local_cfg().disable_metrics()) {
+        _metrics.add_group(
+          prometheus_sanitize::metrics_name("tx:coordinator"),
+          setup.template
+          operator()<ss::metrics::impl::metric_definition_impl, true>({}),
+          {},
+          {});
```

Comment:
It looks like you want to have the metrics aggregation be unconditional (.i.e.: allways aggregate those labels independent of the `aggregate_metrics` flag) which is fine.

However I think if using this `add_group` overload (with support for changing `aggregate_metrics` at runtime) you would need to specify the labels for both "aggregated" and "non_aggregated" labels. Otherwise changing the flag might remove all the aggregation (I think that would happen but didn't test).

Alternatively just use the standard `add_group`.